### ISSUE TITLE
`node.language()` causes panic

### ISSUE THREAD

**Issue by Sarrus1** [silvanshade/tree-sitter-facade#54]
Hello,

I am new to Rust, sorry if this does not have enough details, feel free to let know if I can improve it 👍 

Calling the `language()` method on a `Node` object seem to cause a panic. Because of this, some methods, such as `children_by_field_name` cause a panic.

**Comment by silvanshade** [comment]
I believe this is probably because you need to call `web_tree_sitter_sys::Parser::init()` first. This is an unfortunate artifact of the way the `web-tree-sitter` library (which `tree-sitter-facade` uses when compiling to `wasm32`) is designed. The emscripten machinery used to compile `tree-sitter` to WASM for `web-tree-sitter` requires an explicit initialization before doing anything else.

You can see what I mean on the documentation for the [web-tree-sitter](https://www.npmjs.com/package/web-tree-sitter) package, for example:

```js
const Parser = require('web-tree-sitter');
Parser.init().then(() => { /* the library is ready */ });
```

This detail is not directly exposed or documented in `tree-sitter-facade`, but it probably should be. The solution is to also import the [web-tree-sitter-sys](https://crates.io/crates/web-tree-sitter-sys) crate and manually call [Parser::init](https://docs.rs/web-tree-sitter-sys/latest/web_tree_sitter_sys/struct.Parser.html#method.init).

You can see a complete example where I do that here: https://github.com/silvanshade/tower-lsp-web-demo/blob/2c4f00f44f46d98a067150c35119f27ae7d2236c/crates/browser/src/lib.rs#L43-L45

Note that I am currently experimenting with a redesign of both `web-tree-sitter-sys` and `tree-sitter-facade` (not yet published) that might handle this detail automatically but for now this is how to do it.

**Comment by Sarrus1** [comment]
Ah I see, this makes sense. Thanks!

I could document the behaviour in #55 if you want 👍

**Comment by Sarrus1** [comment]
Unfortunately this did not work, I still get the same panic when using a method that calls `node.language()`, as you can see here:
https://github.com/Sarrus1/SPFormat/blob/68366a0698adf7b3a0bdadd56e9a2628c02a27f1/src/main.rs#L32-L36
![Capture](https://user-images.githubusercontent.com/63302440/177972015-4b620a1b-028d-4492-93cc-d67ce84a67e3.PNG)

**Comment by silvanshade** [comment]
Thanks for the details. From a cursory glance at the code I'm not sure why this is happening. I'll try to do some investigation.

One small point though, you probably want to move the call to `Parser::init` from `format_string` into the `main` function. This is because `Parser::init` is only meant to be called once for the entire program. Technically, I think calling it more than once won't cause the program to crash or anything but I'm not 100% sure that it does not have some other unintended consequences.

**Comment by silvanshade** [comment]
I published a new version of the crate `0.9.0` which I believe might fix this issue although I haven't had a chance to properly test it. Please let me know if that works.

Note the API has changed somewhat. You can see a recent implementation using the new version at https://github.com/silvanshade/tower-lsp-web-demo

**Comment by Sarrus1** [comment]
Late response, but after some tests, 0.9.0 has fixed the issue for me 👍 

If needed, here is the commit with the appropriate changes on my project: https://github.com/Sarrus1/SPFormat/commit/5bdbf119690e5c47a1d3a3ffca3987ede6ec4840

Thanks you for the fix 🚀
